### ISSUE TITLE
Offer a fix-it (to insert an include) for unresolved symbols in std::

### ISSUE THREAD

**Issue by ChristianKuehnel** [clangd/clangd#618]
When using something from the STL like `std::unordered_map` it would be nice if clangd automatically inserts a `#include <unordered_map>`into the beginning of the file. This way I would not have to 1) figure out the name of the header and 2) add the `#include` manually.

Especially as I'm not that familiar with the STL, this would save me a few Google searches...

Some more examples I was using recently:
```
std::cout   --> #include <iostream>
std::regex  --> #include <regex>
std::vector --> #include <vector>
std::string --> #include <string>
std::max    --> #include <algorithm>
```

**Comment by HighCommander4** [comment]
This does work in response to code completion, e.g. if you type `std::unord` and then invoke code completion and select `unordered_map`, the include is inserted.

Are you suggesting this be implemented for some other trigger than code completion, or is it not working for you with code completion?

**Comment by ChristianKuehnel** [comment]
VSCode does not show me any code completions for the prefix `std::unord` unless I have that header already included or had it opened before in the same session.

**Comment by HighCommander4** [comment]
Ah, I see, thanks. This is a known issue, tracked in #403.

**Comment by sam-mccall** [comment]
There's another possible approach without indexing the standard library as in #403: we could use our symbol->header map for stdlib in the include-fixer lookup directly.

Indexing it would be better though, if feasible, as it would make code completion work too.

**Comment by HighCommander4** [comment]
Could the symbols in that map potentially be a source of bare-bones completion proposals inside `std::` as well?

**Comment by sam-mccall** [comment]
We only have the symbol names, with no signatures, so you *could* but it might be confusing.

**Comment by HighCommander4** [comment]
Anyways, I think it makes sense to reopen this as there is something distinct from #403 to potentially track here.

**Comment by HighCommander4** [comment]
I wanted to circle back to this to say that, in addition to inserting the include during code completion, clangd does also surface a quick-fix that you can apply after the fact, to insert the include.

Howerver, it does still require that _some_ file in the project already include the header such that clangd has it indexed. That should be fixed in #403, which has a [patch](https://reviews.llvm.org/D105177), so I don't think there's anything further to track here. (Please let me know if I've missed something.)
